### DISCUSSION TITLE
How to sort or group check-results by rules?

### DISCUSSION THREAD

**Discussion by solzard** [astral-sh/ruff#3594] [Q&A]
`ruff check --format grouped` groups found errors by files. How can I do this by error codes?

```
# Expected result
PTH123
foo.py:1:1 ...
bar:py:2:2 ...

N806
foo.py:3:3 ...
bar.py:4:4 ...
```

**Comment by charliermarsh** [comment]
This isn't currently possible, though we could consider supporting it. Do you know of any other linters that support structuring the output by code, as precedent?

**Reply by solzard** [reply]
None has this feature as far as I know.
After migrating to `ruff` on personal projects, I enabled as many rules as possible besides E/F. 
As a natural consequence, many errors show up, and I wanted to have the rule-based grouping to tackle the same violations at once.

Currently, I use the following script `ruff-by-rules.sh`:

```sh
#!/usr/bin/env bash
set -euo pipefail

function main() {
    local target="${1:-.}"
    local rule
    for rule in $(ruff check "$target" --statistics | sort --reverse | awk '{print $2}'); do
        echo "$rule"
        ruff check --select "$rule" --exit-zero "$target"
        echo
    done

}

main "$@"
```

Example usages:
```sh
bash ruff-by-rules.sh
bash ruff-by-rules.sh ./apps/detector
```
